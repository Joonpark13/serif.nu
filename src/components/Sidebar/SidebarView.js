import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import { withStyles } from '@material-ui/core/styles';
import SearchContainer from './search/SearchContainer';
import CartContainer from './cart/CartContainer';
import BrowseContainer from './browse/BrowseContainer';

export const styles = {
  wrapper: {
    height: 'calc(100vh - 64px)', // 64px is height of TopBar
    overflow: 'auto',
  },
  tabWidth: {
    minWidth: 0,
  },
};

class SidebarView extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: 'search',
    };

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event, value) {
    this.setState({ value });
  }

  render() {
    const { classes } = this.props;
    const { value } = this.state;

    return (
      <div className={classes.wrapper}>
        <Tabs
          value={value}
          textColor="primary"
          indicatorColor="primary"
          onChange={this.handleChange}
          variant="fullWidth"
        >
          <Tab value="search" label="Search" className={classes.tabWidth} />
          <Tab value="browse" label="Browse" className={classes.tabWidth} />
          <Tab value="cart" label="Cart" className={classes.tabWidth} />
        </Tabs>
        {value === 'search' && <SearchContainer />}
        {value === 'browse' && <BrowseContainer />}
        {value === 'cart' && <CartContainer />}
      </div>
    );
  }
}

SidebarView.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export { SidebarView as UnstyledSidebarView };
export default withStyles(styles)(SidebarView);
