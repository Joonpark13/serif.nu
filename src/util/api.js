/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
