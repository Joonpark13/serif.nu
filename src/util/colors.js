export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
