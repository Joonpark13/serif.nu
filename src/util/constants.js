/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
