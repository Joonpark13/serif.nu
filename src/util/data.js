import termsData from 'data/terms.json';

/* eslint-disable import/prefer-default-export */
/* istanbul ignore next */
export const currentTerm = termsData.reduce((highestTerm, term) => {
  if (term.id > highestTerm.id) {
    return term;
  }
  return highestTerm;
});
/* eslint-enable import/prefer-default-export */
