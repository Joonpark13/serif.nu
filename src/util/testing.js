/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
