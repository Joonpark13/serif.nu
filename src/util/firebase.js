import firebase from 'firebase/app';
import 'firebase/firestore';
import 'firebase/auth';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export const db = firebase.firestore();
export const auth = firebase.auth();
