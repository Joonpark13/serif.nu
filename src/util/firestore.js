import firebase from 'firebase/app';
import 'firebase/firestore';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export default firebase.firestore();
