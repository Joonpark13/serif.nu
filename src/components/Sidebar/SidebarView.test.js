import React from 'react';
import { shallow } from 'enzyme';
import { mockStyles } from 'util/testing';
import { UnstyledSidebarView, styles } from './SidebarView';

describe('SidebarView', () => {
  it('renders correctly', () => {
    const classes = mockStyles(styles);
    const wrapper = shallow(<UnstyledSidebarView
      classes={classes}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('changes tabs correctly', () => {
    const classes = mockStyles(styles);
    const wrapper = shallow(<UnstyledSidebarView
      classes={classes}
    />);
    wrapper.instance().handleChange('onChange', 'browse');

    expect(wrapper).toMatchSnapshot();
  });
});
