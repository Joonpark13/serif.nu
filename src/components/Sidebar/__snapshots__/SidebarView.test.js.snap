// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarView changes tabs correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SidebarView
    classes={
      Object {
        "tabWidth": "",
        "wrapper": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Tabs)
          indicatorColor="primary"
          onChange={[Function]}
          textColor="primary"
          value="browse"
          variant="fullWidth"
        >
          <WithStyles(Tab)
            className=""
            label="Search"
            value="search"
          />
          <WithStyles(Tab)
            className=""
            label="Browse"
            value="browse"
          />
          <WithStyles(Tab)
            className=""
            label="Cart"
            value="cart"
          />
        </WithStyles(Tabs)>,
        false,
        <Connect(Browse) />,
        false,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Tab)
              className=""
              label="Search"
              value="search"
            />,
            <WithStyles(Tab)
              className=""
              label="Browse"
              value="browse"
            />,
            <WithStyles(Tab)
              className=""
              label="Cart"
              value="cart"
            />,
          ],
          "indicatorColor": "primary",
          "onChange": [Function],
          "textColor": "primary",
          "value": "browse",
          "variant": "fullWidth",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "label": "Search",
              "value": "search",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "label": "Browse",
              "value": "browse",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "label": "Cart",
              "value": "cart",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Tabs)
            indicatorColor="primary"
            onChange={[Function]}
            textColor="primary"
            value="browse"
            variant="fullWidth"
          >
            <WithStyles(Tab)
              className=""
              label="Search"
              value="search"
            />
            <WithStyles(Tab)
              className=""
              label="Browse"
              value="browse"
            />
            <WithStyles(Tab)
              className=""
              label="Cart"
              value="cart"
            />
          </WithStyles(Tabs)>,
          false,
          <Connect(Browse) />,
          false,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Tab)
                className=""
                label="Search"
                value="search"
              />,
              <WithStyles(Tab)
                className=""
                label="Browse"
                value="browse"
              />,
              <WithStyles(Tab)
                className=""
                label="Cart"
                value="cart"
              />,
            ],
            "indicatorColor": "primary",
            "onChange": [Function],
            "textColor": "primary",
            "value": "browse",
            "variant": "fullWidth",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "label": "Search",
                "value": "search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "label": "Browse",
                "value": "browse",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "label": "Cart",
                "value": "cart",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`SidebarView changes to cart correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SidebarView
    classes={
      Object {
        "tabWidth": "",
        "wrapper": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Tabs)
          indicatorColor="primary"
          onChange={[Function]}
          textColor="primary"
          value="cart"
          variant="fullWidth"
        >
          <WithStyles(Tab)
            className=""
            label="Search"
            value="search"
          />
          <WithStyles(Tab)
            className=""
            label="Browse"
            value="browse"
          />
          <WithStyles(Tab)
            className=""
            label="Cart"
            value="cart"
          />
        </WithStyles(Tabs)>,
        false,
        false,
        <Connect(Component) />,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Tab)
              className=""
              label="Search"
              value="search"
            />,
            <WithStyles(Tab)
              className=""
              label="Browse"
              value="browse"
            />,
            <WithStyles(Tab)
              className=""
              label="Cart"
              value="cart"
            />,
          ],
          "indicatorColor": "primary",
          "onChange": [Function],
          "textColor": "primary",
          "value": "cart",
          "variant": "fullWidth",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "label": "Search",
              "value": "search",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "label": "Browse",
              "value": "browse",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "label": "Cart",
              "value": "cart",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Tabs)
            indicatorColor="primary"
            onChange={[Function]}
            textColor="primary"
            value="cart"
            variant="fullWidth"
          >
            <WithStyles(Tab)
              className=""
              label="Search"
              value="search"
            />
            <WithStyles(Tab)
              className=""
              label="Browse"
              value="browse"
            />
            <WithStyles(Tab)
              className=""
              label="Cart"
              value="cart"
            />
          </WithStyles(Tabs)>,
          false,
          false,
          <Connect(Component) />,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Tab)
                className=""
                label="Search"
                value="search"
              />,
              <WithStyles(Tab)
                className=""
                label="Browse"
                value="browse"
              />,
              <WithStyles(Tab)
                className=""
                label="Cart"
                value="cart"
              />,
            ],
            "indicatorColor": "primary",
            "onChange": [Function],
            "textColor": "primary",
            "value": "cart",
            "variant": "fullWidth",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "label": "Search",
                "value": "search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "label": "Browse",
                "value": "browse",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "label": "Cart",
                "value": "cart",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`SidebarView renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SidebarView
    classes={
      Object {
        "tabWidth": "",
        "wrapper": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Tabs)
          indicatorColor="primary"
          onChange={[Function]}
          textColor="primary"
          value="search"
          variant="fullWidth"
        >
          <WithStyles(Tab)
            className=""
            label="Search"
            value="search"
          />
          <WithStyles(Tab)
            className=""
            label="Browse"
            value="browse"
          />
          <WithStyles(Tab)
            className=""
            label="Cart"
            value="cart"
          />
        </WithStyles(Tabs)>,
        <Connect(Component) />,
        false,
        false,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Tab)
              className=""
              label="Search"
              value="search"
            />,
            <WithStyles(Tab)
              className=""
              label="Browse"
              value="browse"
            />,
            <WithStyles(Tab)
              className=""
              label="Cart"
              value="cart"
            />,
          ],
          "indicatorColor": "primary",
          "onChange": [Function],
          "textColor": "primary",
          "value": "search",
          "variant": "fullWidth",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "label": "Search",
              "value": "search",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "label": "Browse",
              "value": "browse",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "label": "Cart",
              "value": "cart",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Tabs)
            indicatorColor="primary"
            onChange={[Function]}
            textColor="primary"
            value="search"
            variant="fullWidth"
          >
            <WithStyles(Tab)
              className=""
              label="Search"
              value="search"
            />
            <WithStyles(Tab)
              className=""
              label="Browse"
              value="browse"
            />
            <WithStyles(Tab)
              className=""
              label="Cart"
              value="cart"
            />
          </WithStyles(Tabs)>,
          <Connect(Component) />,
          false,
          false,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Tab)
                className=""
                label="Search"
                value="search"
              />,
              <WithStyles(Tab)
                className=""
                label="Browse"
                value="browse"
              />,
              <WithStyles(Tab)
                className=""
                label="Cart"
                value="cart"
              />,
            ],
            "indicatorColor": "primary",
            "onChange": [Function],
            "textColor": "primary",
            "value": "search",
            "variant": "fullWidth",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "label": "Search",
                "value": "search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "label": "Browse",
                "value": "browse",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "label": "Cart",
                "value": "cart",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
